### ISSUE TITLE
Connecting two non-overlapping meshes

### ISSUE THREAD

**Issue by 1234shel** [mikedh/trimesh#874]
Hi,

I wanted to know if it was possible to connect two non-overlapping meshes using specific faces? I have looked into boolean union and concatenate functions and I don't quite get the result I am looking for. I want to connect two meshes using specific surfaces so as to preserve their individual shapes and volumes. Is it possible?

**Comment by bigeyesung** [comment]
Not sure if I get it but it sounds like my questions as well. So far I am using ray-tracing from one face to the other. Using the face normal vector as the ray. If it hits then we say these meshes are connected

**Comment by 1234shel** [comment]
I figured out how to do this. I use the stable posses function to identify appropriate faces and then use the given transformations to align the faces and then do a boolean union.

**Comment by bigeyesung** [comment]
which function you are using? compute_stable_poses()?

**Comment by 1234shel** [comment]
Yup. It will give you a transformation against the z-axis for both meshes. You can transpose one of the meshes against the z-axis, align the faces and do a union. With SCAD as the engine you can get rid of the internal geometry and hence you have a single mesh combining the two.
